src/main/scala/com/fulcrumgenomics/phasing/HapCutToVcf.scala

```diff
@@ -0,0 +1,413 @@
+    val phasedWriter   = makeWriter(output, vcfReader, addPhasedHeaderLines=true)
+    val unphasedWriter = makeWriter(unphasedOutput, vcfReader, addPhasedHeaderLines=false)
+
+    val iterator = new MergingIterator(input, vcfReader, readBackedPhasingFormat)
+    iterator.foreach {
+      case Left(ctx)  => phasedWriter.add(ctx)
+      case Right(ctx) => unphasedWriter.add(ctx)
+    }
+
+    CloserUtil.close(vcfReader)
+    phasedWriter.close()
+    unphasedWriter.close()
+  }
+
+  /** Creates a VCF writer, adding extra header lines if the output is the phased VCF. */
+  private def makeWriter(path: PathToVcf, vcfReader: VCFFileReader, addPhasedHeaderLines: Boolean): VariantContextWriter = {
+    val inputHeader = vcfReader.getFileHeader
+
+    val builder = new VariantContextWriterBuilder()
+      .setOutputFile(path.toFile)
+      .setReferenceDictionary(inputHeader.getSequenceDictionary)
+      .setOption(Options.INDEX_ON_THE_FLY)
+      .modifyOption(Options.ALLOW_MISSING_FIELDS_IN_HEADER, allowMissingFieldsInVcfHeader)
+    val writer: VariantContextWriter = builder.build
+
+    val outHeader = if (addPhasedHeaderLines) {
+      // get the header lines to skip
+      val headerLinesToSkip = HapCutCallGenotypeInfo.formatHeaderKeys
+        .filter { key => inputHeader.hasFormatLine(key) }
+        .map { key => inputHeader.getFormatHeaderLine(key) }
+      // get all the header lines from the input
+      val headerLines: util.Set[VCFHeaderLine] = new util.HashSet[VCFHeaderLine](
+        inputHeader.getMetaDataInSortedOrder.filter { line =>
+          if (headerLinesToSkip.contains(line)) {
+            logger.debug(s"Will overwrite header line: '${line.toString}")
+            false
+          }
+          else true
+        }.toSet
+      )
+      // add standard header lines
+      VCFStandardHeaderLines.addStandardFormatLines(headerLines, false, Genotype.PRIMARY_KEYS)
+      // create the header
+      val header: VCFHeader = new VCFHeader(headerLines, inputHeader.getSampleNamesInOrder)
+      // add the new header lines
+      HapCutCallGenotypeInfo.formatHeaderLines.foreach(header.addMetaDataLine)
+      header
+    }
+    else {
+      inputHeader
+    }
+    writer.writeHeader(outHeader)
+    writer
+  }
+}
+
+/** An iterator over all variants in the provided VCFFileReader.  The `input` is the output of HapCut.
+  *
+  * If the variant in the VCFFileReader was not phased by HapCut, then we will return Right(VariantContext).  If the
+  * variant was phased by HapCut, we will return Left(VariantContext).
+  */
+private class MergingIterator(input: FilePath, vcfReader: VCFFileReader, readBackedPhasingFormat: Boolean) extends Iterator[Either[VariantContext,VariantContext]] {
+
+  private val sourceIterator = vcfReader.toStream.map { ctx =>
+    // NB: unsets the phase in case it was already set
+    new VariantContextBuilder(ctx).genotypes(new GenotypeBuilder(ctx.getGenotype(0)).phased(false).make()).make()
```

Comment:
Suppose you are trying to re-phase the truth data, rather than rely on your own calls, then this makes a lot of sense to unphase, so it is likely "at some point in the future it is 99% of the time" but your point is well taken.